panic: runtime error: invalid memory address or nil pointer dereference

```
May 28 01:25:13.322164 INF Closing leader controller component=leader-controller namespace=default shard=1 term=2
PASS
coverage: 59.3% of statements
May 28 01:25:14.047375 WRN Session expired client-identity=my-identity component=session namespace=default session-id=0 shard=2
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x15e90bb]

goroutine 1452 [running]:
github.com/streamnative/oxia/server.(*leaderController).list.func1()
	/home/runner/work/oxia/oxia/server/leader_controller.go:684 +0x23b
github.com/streamnative/oxia/common.DoWithLabels.func1({0x1cceec8?, 0xc001078300?})
	/home/runner/work/oxia/oxia/common/pprof.go:46 +0x2f
runtime/pprof.Do({0x1ccec28, 0x356b1e0}, {{0xc000c21b00?, 0xc00098aee0?, 0x0?}}, 0xc00098af70)
	/opt/hostedtoolcache/go/1.24.3/x64/src/runtime/pprof/runtime.go:51 +0x118
github.com/streamnative/oxia/common.DoWithLabels({0x1ccec28, 0x356b1e0}, 0xc001149260, 0xc00112d900)
	/home/runner/work/oxia/oxia/common/pprof.go:42 +0x337
created by github.com/streamnative/oxia/server.(*leaderController).list in goroutine 1451
	/home/runner/work/oxia/oxia/server/leader_controller.go:668 +0x485
FAIL	github.com/streamnative/oxia/server	8.486s
	github.com/streamnative/oxia/server/auth		coverage: 0.0% of statements
ok  	github.com/streamnative/oxia/server/kv	8.845s	coverage: 79.4% of statements
ok  	github.com/streamnative/oxia/server/util	1.019s	coverage: 15.6% of statements
	github.com/streamnative/oxia/server/util/crc		coverage: 0.0% of statements
ok  	github.com/streamnative/oxia/server/wal	5.326s	coverage: 75.4% of statements
ok  	github.com/streamnative/oxia/server/wal/codec	1.014s	coverage: 83.6% of statements
ok  	github.com/streamnative/oxia/tests/security/auth	5.192s	coverage: [no statements]
ok  	github.com/streamnative/oxia/tests/security/tls	10.479s	coverage: [no statements]
ok  	github.com/streamnative/oxia/tests/sequence	1.090s	coverage: [no statements]
```

--- a/server/session.go
+++ b/server/session.go
@@ -172,7 +172,6 @@ func (s *session) waitForHeartbeats() {
 		s.latch.Done()
 	}()
 	for {
-
 		select {
 		case <-s.ctx.Done():
 			return

